pkg/kncloudevents/event_dispatcher.go

```diff
@@ -319,6 +343,16 @@ func createRequest(ctx context.Context, message binding.Message, target duckv1.A
 		request.Header[key] = val
 	}
 
+	if oidcServiceAccount != nil {
+		if target.Audience != nil && *target.Audience != "" {
+			jwt, err := d.oidcTokenProvider.GetJWT(*oidcServiceAccount, *target.Audience)
+			if err != nil {
+				return nil, fmt.Errorf("could not get JWT: %w", err)
+			}
+			request.Header.Set("Authorization", fmt.Sprintf("Bearer: %s", jwt))
```

Comment:
@creydr sorry if this is a dumb question, I still don't 100% understand the full OIDC flow. In this code, why do we fail if we can't get a JWT, instead of just sending an event without an authorization header?

Reply:
I see it as a contract between the caller and the dispatcher here. When the caller set an oidcServiceAccount (via the `WithOIDCAuthentication(sa)` option), we assume that they want to use authentication. If we fail to request a token then, then we should report this back to the caller IMO.

Reply:
Does this help?

Reply:
Yeah that makes sense to me!

Reply:
Thanks for clarifying @creydr !!